Recollect metadata fail if plugin was previously compiled with older JDK

### Plugin Modernizer version

1046.v7cedb_6c8df3f

### What Operating System are you using

Linux

### Reproduction steps

plugin-modernizer dry-run --plugins rebuild --recipe UpgradeToRecommendCoreVersion

```
Using minimum JDK 11 from metadata for plugin rebuild 
Skipping formatting for plugin rebuild as it is not using Spotless 
Verifying plugin rebuild with JDK 11... Please be patient 
Done 
Plugin rebuild verified successfully with JDK 11 
Collecting metadata for plugin rebuild... Please be patient 
Build failed 
Done 
Object does not exist in source cache at /home/valentin/.cache/jenkins-plugin-modernizer-cli/rebuild/sources/target/./plugin-metadata 
{"key":"plugin-metadata","path":"."} 
```

The plugin builds well with JDK 11 but fail after when we recollect metrics.

```
2024-12-29T05:39:18.609Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] <<< rewrite:5.47.1:run (default-cli) < process-test-classes @ rebuild <<< 
2024-12-29T05:39:18.609Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO]  
2024-12-29T05:39:18.624Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO]  
2024-12-29T05:39:18.625Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] --- rewrite:5.47.1:run (default-cli) @ rebuild --- 
2024-12-29T05:39:18.795Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] Using active recipe(s) [io.jenkins.tools.pluginmodernizer.FetchMetadata] 
2024-12-29T05:39:18.795Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] Using active styles(s) [] 
2024-12-29T05:39:20.603Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] ------------------------------------------------------------------------ 
2024-12-29T05:39:20.603Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] BUILD FAILURE 
2024-12-29T05:39:20.603Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] ------------------------------------------------------------------------ 
2024-12-29T05:39:20.603Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] Total time:  5.282 s 
2024-12-29T05:39:20.603Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] Finished at: 2024-12-29T06:39:20+01:00 
2024-12-29T05:39:20.603Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [INFO] ------------------------------------------------------------------------ 
2024-12-29T05:39:20.604Z [INFO] [Thread=StreamPumper-systemOut] - i.j.t.p.core.impl.MavenInvoker # [ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:5.47.1:run (default-cli) on project rebuild: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:5.47.1:run failed: Could not load class io.jenkins.tools.pluginmodernizer.core.recipes.FetchMetadata : java.lang.UnsupportedClassVersionError: io/jenkins/tools/pluginmodernizer/core/recipes/FetchMetadata has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0 -> [Help 1] 

```

### Expected Results

No issue.

Will do a PR.

Safest is to perform a "clean" phase just before the recollection to cleanup ".class" files on the target repo

--- a/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java
+++ b/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java
@@ -290,6 +290,8 @@ public class PluginModernizer {
 
             // Recollect metadata after modernization
             if (!config.isFetchMetadataOnly()) {
+                plugin.withJDK(JDK.JAVA_17);
+                plugin.clean(mavenInvoker);
                 collectMetadata(plugin);
                 LOG.debug(
                         "Plugin {} metadata after modernization: {}",

